### PR TITLE
Renamed procs/jobs to steps in code

### PR DESCRIPTION
**Pull request by Harikesh00**
Renamed `procs` to `steps` in code for the issue #1288

### CODE REVIEW

**Comment by Harikesh00** [comment]
@6543 plz review

File: cmd/agent/flags.go [hunk 1]
```diff
@@ -79,8 +79,8 @@ var flags = []cli.Flag{
 		Usage:   "List of labels to filter tasks on. An agent must be assigned every tag listed in a task to be selected.",
 	},
 	&cli.IntFlag{
-		EnvVars: []string{"WOODPECKER_MAX_PROCS"},
-		Name:    "max-procs",
+		EnvVars: []string{"WOODPECKER_MAX_STEPS"},
```
Comment (6543) [inline cmd/agent/flags.go:82]:
```suggestion
		EnvVars: []string{"WOODPECKER_MAX_STEPS", "WOODPECKER_MAX_PROCS"},
```

File: server/model/log.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ package model
 
 type Logs struct {
 	ID     int64  `xorm:"pk autoincr 'log_id'"`
-	ProcID int64  `xorm:"UNIQUE 'log_job_id'"`
+	StepID int64  `xorm:"UNIQUE 'log_job_id'"`
```
Comment (6543) [inline server/model/log.go:19]:
```suggestion
	StepID int64  `xorm:"UNIQUE 'log_step_id'"`
```
+migration

**Review by 6543 — CHANGES_REQUESTED** [review]

File: agent/runner.go [hunk 3]
```diff
@@ -186,7 +186,7 @@ func (r *Runner) Run(ctx context.Context) error {
 
 		file := &rpc.File{
 			Mime: "application/json+logs",
-			Proc: proc.Alias,
+			Step: step.Alias,
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply agent/runner.go:189]: reverting

File: agent/runner.go [hunk 4]
```diff
@@ -250,35 +250,35 @@ func (r *Runner) Run(ctx context.Context) error {
...
 			Str("stage", state.Pipeline.Step.Alias).
 			Err(state.Process.Error).
 			Int("exit_code", state.Process.ExitCode).
 			Bool("exited", state.Process.Exited).
 			Logger()
 
-		procState := rpc.State{
-			Proc:     state.Pipeline.Step.Alias,
+		stepState := rpc.State{
+			Step:     state.Pipeline.Step.Alias,
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply agent/runner.go:262]: reverting

File: cli/exec/line.go [hunk 5]
```diff
@@ -32,7 +32,7 @@ const (
 
 // Line is a line of console output.
 type Line struct {
-	Proc string `json:"proc,omitempty"`
+	Step string `json:"step,omitempty"`
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply cli/exec/line.go:35]: reverting

File: cli/pipeline/ps.go [hunk 6]
```diff
@@ -76,8 +76,8 @@ func pipelinePs(c *cli.Context) error {
 		return err
 	}
 
-	for _, proc := range pipeline.Procs {
-		for _, child := range proc.Children {
+	for _, step := range pipeline.Steps {
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply cli/pipeline/ps.go:79]: reverting

File: docs/versioned_docs/version-0.15/30-administration/00-setup.md [hunk 7]
```diff
@@ -8,7 +8,7 @@ A Woodpecker deployment consists of two parts:
 >
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply docs/versioned_docs/version-0.15/30-administration/00-setup.md:8]: reverted

File: pipeline/backend/local/local.go [hunk 8]
```diff
@@ -55,33 +55,33 @@ func (e *local) Load() error {
 }
 
 // Setup the pipeline environment.
-func (e *local) Setup(ctx context.Context, proc *types.Config) error {
+func (e *local) Setup(ctx context.Context, step *types.Config) error {
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply pipeline/backend/local/local.go:58]: done

File: pipeline/backend/ssh/ssh.go [hunk 9]
```diff
@@ -84,33 +84,33 @@ func (e *ssh) Load() error {
 }
 
 // Setup the pipeline environment.
-func (e *ssh) Setup(ctx context.Context, proc *types.Config) error {
+func (e *ssh) Setup(ctx context.Context, step *types.Config) error {
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply pipeline/backend/ssh/ssh.go:87]: done

File: pipeline/rpc/line.go [hunk 10]
```diff
@@ -22,7 +22,7 @@ const (
 
 // Line is a line of console output.
 type Line struct {
-	Proc string `json:"proc,omitempty"`
+	Step string `json:"step,omitempty"`
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply pipeline/rpc/line.go:25]: ?
didn't get u

File: pipeline/tracer.go [hunk 11]
```diff
@@ -14,7 +14,7 @@ type Tracer interface {
 // functions as a Tracer.
 type TraceFunc func(*State) error
 
-// Trace calls f(proc, state).
+// Trace calls f(step, state).
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply pipeline/tracer.go:17]: done

File: server/model/file.go [hunk 12]
```diff
@@ -20,16 +20,16 @@ import "io"
...
+	FileRead(*Step, string) (io.ReadCloser, error)
 	FileCreate(*File, io.Reader) error
 }
 
 // File represents a pipeline artifact.
 type File struct {
 	ID         int64  `json:"id"      xorm:"pk autoincr 'file_id'"`
 	PipelineID int64  `json:"-"       xorm:"INDEX 'file_pipeline_id'"`
-	ProcID     int64  `json:"proc_id" xorm:"UNIQUE(s) INDEX 'file_proc_id'"`
+	StepID     int64  `json:"step_id" xorm:"UNIQUE(s) INDEX 'file_step_id'"`
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply server/model/file.go:32]: should I revert this change 
or can I write a migration script?
if yes any reference for it, from where I can follow

File: server/store/datastore/migration/000_legacy_to_xorm.go [hunk 13]
```diff
@@ -45,13 +45,13 @@ var legacy2Xorm = task{
 			"create-table-builds",
 			"create-index-builds-repo",
 			"create-index-builds-author",
-			// procs
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply server/store/datastore/migration/000_legacy_to_xorm.go:48]: reverted

File: server/store/datastore/migration/000_legacy_to_xorm.go [hunk 14]
```diff
@@ -128,9 +128,9 @@ var legacy2Xorm = task{
 				"DROP INDEX IF EXISTS ix_build_status_running;",
 				"DROP INDEX IF EXISTS ix_build_repo;",
 				"DROP INDEX IF EXISTS ix_build_author;",
-				"DROP INDEX IF EXISTS proc_build_ix;",
+				"DROP INDEX IF EXISTS step_build_ix;",
 				"DROP INDEX IF EXISTS file_build_ix;",
-				"DROP INDEX IF EXISTS file_proc_ix;",
+				"DROP INDEX IF EXISTS file_step_ix;",
```
[the remark this answers is not in the document]
Comment (Harikesh00) [reply server/store/datastore/migration/000_legacy_to_xorm.go:133]: reverted

File: cli/pipeline/ps.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (6543) [reply cli/pipeline/ps.go:79]: well you can do it in this batch but then you need migrations

[the remark this answers is not in the document]
Comment (Harikesh00) [reply cli/pipeline/ps.go:79]: ok, I'll write the migration
can I get any reference for it?
from where I can follow what all changes need to do & make the changes

[the remark this answers is not in the document]
Comment (6543) [reply cli/pipeline/ps.go:79]: https://github.com/woodpecker-ci/woodpecker/pull/1296

File: docs/versioned_docs/version-0.15/30-administration/00-setup.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (6543) [reply docs/versioned_docs/version-0.15/30-administration/00-setup.md:8]: @Harikesh00 pull did not get new commits, did you forgot to push?

[the remark this answers is not in the document]
Comment (Harikesh00) [reply docs/versioned_docs/version-0.15/30-administration/00-setup.md:8]: haven't pushed yet
waiting for confirmation on migration changes.
do I cover migration changes in the same MR
or should I revert the migration changes & will cover in next MR?

[the remark this answers is not in the document]
Comment (6543) [reply docs/versioned_docs/version-0.15/30-administration/00-setup.md:8]: well just do it in here ... dont force push just append commits, we squashmerge

File: server/model/file.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (6543) [reply server/model/file.go:32]: https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md#code-review

[the remark this answers is not in the document]
Comment (6543) [reply server/model/file.go:32]: just migrate

File: docs/versioned_docs/version-0.15/30-administration/00-setup.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (Harikesh00) [reply docs/versioned_docs/version-0.15/30-administration/00-setup.md:8]: ok sure
I'll inform here once changes push

**Comment by Harikesh00** [comment]
@6543 @qwerty287  all breaking changes that are commented on PR are because of migration?
if I add the migration script then can I keep those changes? (no need to revert)

File: server/model/file.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (Harikesh00) [reply server/model/file.go:32]: ok

File: cmd/agent/flags.go (hunk 1 above)
Comment (Harikesh00) [reply cmd/agent/flags.go]: done

**Comment by Harikesh00** [comment]
@qwerty287 @6543 migration script added & review comments resolved
plz review

**Comment by Harikesh00** [comment]
I didn't get what fallback means, why it's required and what it will do
can u plz explain

File: cmd/agent/flags.go [hunk 15]
```diff
@@ -79,8 +79,8 @@ var flags = []cli.Flag{
 		Usage:   "List of labels to filter tasks on. An agent must be assigned every tag listed in a task to be selected.",
 	},
 	&cli.IntFlag{
-		EnvVars: []string{"WOODPECKER_MAX_PROCS"},
-		Name:    "max-procs",
+		EnvVars: []string{"WOODPECKER_MAX_STEPS", "WOODPECKER_MAX_PROCS"},
```
Comment (6543) [inline cmd/agent/flags.go:82]: well still that's not steps but workflows ...

File: agent/runner.go [hunk 16]
```diff
@@ -67,7 +67,7 @@ func (r *Runner) Run(ctx context.Context) error {
 	meta, _ := metadata.FromOutgoingContext(ctx)
 	ctxmeta := metadata.NewOutgoingContext(context.Background(), meta)
 
-	// get the next job from the queue
+	// get the next pipeline from the queue
```
Comment (6543) [inline agent/runner.go:70]:
```suggestion
	// get the next workflow from the queue
```

File: cli/exec/flags.go [hunk 17]
```diff
@@ -260,8 +260,8 @@ var flags = []cli.Flag{
 		Name:    "prev-commit-author-email",
 	},
 	&cli.IntFlag{
-		EnvVars: []string{"CI_JOB_NUMBER"},
-		Name:    "job-number",
+		EnvVars: []string{"CI_STEP_NUMBER"},
```
Comment (6543) [inline cli/exec/flags.go:263]:
```suggestion
		EnvVars: []string{"CI_STEP_NUMBER", "CI_JOB_NUMBER"},
```

File: docs/docs/30-administration/00-setup.md [hunk 18]
```diff
@@ -9,7 +9,7 @@ A Woodpecker deployment consists of two parts:
 >
 > If you have 4 agents installed and connected to the Woodpecker server, your system will process 4 builds in parallel.
 >
-> You can add more agents to increase the number of parallel builds or set the agent's `WOODPECKER_MAX_PROCS=1` environment variable to increase the number of parallel builds for that agent.
+> You can add more agents to increase the number of parallel steps or set the agent's `WOODPECKER_MAX_WORKFLOWS=1` environment variable to increase the number of parallel builds for that agent.
```
Comment (6543) [inline docs/docs/30-administration/00-setup.md:12]:
```suggestion
> You can add more agents to increase the number of parallel steps or set the agent's `WOODPECKER_MAX_WORKFLOWS=1` environment variable to increase the number of parallel workflows for that agent.
```

File: docs/docs/92-awesome.md [hunk 19]
```diff
@@ -22,7 +22,7 @@ If you have some missing resources, please feel free to [open a pull-request](ht
 
 - [Convert Drone CI pipelines to Woodpecker CI](https://codeberg.org/lafriks/woodpecker-pipeline-transform)
 - [Ansible NAS](https://github.com/davestephens/ansible-nas/) - a homelab Ansible playbook that can set up Woodpecker-CI and Gitea
-- [picus](https://github.com/windsource/picus) - Picus connects to a Woodpecker CI server and creates an agent in the cloud when there are pending jobs.
+- [picus](https://github.com/windsource/picus) - Picus connects to a Woodpecker CI server and creates an agent in the cloud when there are pending steps.
```
Comment (6543) [inline docs/docs/92-awesome.md:25]:
```suggestion
- [picus](https://github.com/windsource/picus) - Picus connects to a Woodpecker CI server and creates an agent in the cloud when there are pending workflows.
```

File: pipeline/tracer.go [hunk 20]
```diff
@@ -37,15 +37,19 @@ var DefaultTracer = TraceFunc(func(state *State) error {
 	state.Pipeline.Step.Environment["CI_BUILD_STARTED"] = strconv.FormatInt(state.Pipeline.Time, 10)
 	state.Pipeline.Step.Environment["CI_BUILD_FINISHED"] = strconv.FormatInt(time.Now().Unix(), 10)
 
+	state.Pipeline.Step.Environment["CI_STEP_STATUS"] = "success"
+	state.Pipeline.Step.Environment["CI_STEP_STARTED"] = strconv.FormatInt(state.Pipeline.Time, 10)
```
Comment (6543) [inline pipeline/tracer.go:41]: can we sort all deprecated blocks below ?

File: server/store/datastore/migration/011_columns_rename_procs_to_steps.go [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+
+package migration
+
+import (
+	"strings"
+
+	"xorm.io/xorm"
+)
+
+var renameTableProcsToSteps = task{
```
Comment (6543) [inline server/store/datastore/migration/011_columns_rename_procs_to_steps.go:23]: could you put them into one migration func ... why do we need different one?

File: cmd/agent/flags.go [hunk 22]
```diff
@@ -79,8 +79,8 @@ var flags = []cli.Flag{
 		Usage:   "List of labels to filter tasks on. An agent must be assigned every tag listed in a task to be selected.",
 	},
 	&cli.IntFlag{
-		EnvVars: []string{"WOODPECKER_MAX_PROCS"},
-		Name:    "max-procs",
+		EnvVars: []string{"WOODPECKER_MAX_WORKFLOWS", "WOODPECKER_MAX_PROCS"},
+		Name:    "max-steps",
```
Comment (6543) [inline cmd/agent/flags.go:83]:
```suggestion
		Name:    "max-workflows",
```

**Comment by 6543** [comment]
diff --git a/web/components.d.ts b/web/components.d.ts
index 8a53c720..0600280b 100644
--- a/web/components.d.ts
+++ b/web/components.d.ts
@@ -75,8 +75,6 @@ declare module '@vue/runtime-core' {
     PipelineItem: typeof import('./src/components/repo/pipeline/PipelineItem.vue')['default']
     PipelineList: typeof import('./src/components/repo/pipeline/PipelineList.vue')['default']
     PipelineLog: typeof import('./src/components/repo/pipeline/PipelineLog.vue')['default']
-    PipelineProcDuration: typeof import('./src/components/repo/pipeline/PipelineProcDuration.vue')['default']
-    PipelineProcList: typeof import('./src/components/repo/pipeline/PipelineProcList.vue')['default']
     PipelineRunningIcon: typeof import('./src/components/repo/pipeline/PipelineRunningIcon.vue')['default']
     PipelineStatusIcon: typeof import('./src/components/repo/pipeline/PipelineStatusIcon.vue')['default']
     PipelineStepDuration: typeof import('./src/components/repo/pipeline/PipelineStepDuration.vue')['default']

File: server/shared/stepStatus.go [hunk 23]
```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 mhmxs.
```
Comment (6543) [inline server/shared/stepStatus.go:1]:
```suggestion
// Copyright 2022 Woodpecker Authors
// Copyright 2019 mhmxs
```

File: server/shared/stepStatus_test.go [hunk 24]
```diff
@@ -0,0 +1,286 @@
+// Copyright 2019 mhmxs.
```
Comment (6543) [inline server/shared/stepStatus_test.go:1]:
```suggestion
// Copyright 2022 Woodpecker Authors
// Copyright 2019 mhmxs
```

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
:point_up:

**Comment by Harikesh00** [comment]
Thank you @6543 @qwerty287 for all your support, you did a lot

**Pull request merged**